File: act08.py
hora = input("Hora: ")

entera = int(hora)
decimal = abs(hora) - abs(int(hora))

hours = entera
minut = decimal * 60

mins = int(minut)
sec = abs(minut) - abs(int(minut))

seca = sec * 60
secs = round(seca, 0)
secsi = int(secs)
if (secsi != 0) and (mins != 0):
    print(str(hours) + " horas " + str(mins) + " minutos y " + str(secsi) + " segundos")
if (secsi == 0) and (mins != 0):
    print(str(hours) + " horas " + str(mins) + " minutos")
if (secsi != 0) and (mins == 0):
    print(str(hours) + " horas y " + str(secsi) + " segundos")
if (secsi == 0) and (mins == 0):
    print(str(hours) + " horas")

File: act08.c
#include <stdio.h>
#include <math.h>
int main(){
  float momento;
  int horas;
  float realmins;
  int minutos;
  int segundos;
  printf("Hora: ");
  scanf("%f", &momento);
  horas = momento;
  realmins = (momento - horas) * 60;
  minutos = realmins;
  segundos = round((realmins - minutos) * 60);
  if(segundos != 0 && minutos != 0){
    printf("%d horas %d minutos y %d segundos.", horas, minutos, segundos);
  }
  else{
    if(segundos == 0 && minutos != 0){
      printf("%d horas y %d minutos.", horas, minutos);
    }
    else{
      if(segundos != 0 && minutos == 0){
        printf("%d horas y %d segundos.", horas, segundos);
      }
      else{
        if(segundos == 0 && minutos == 0){
          printf("%d horas.", horas);
        }
      }
    }
  }
  printf("\n");
  return 0;
}
